src/feeds/ConvexCurvePriceFeed.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity ^0.8.13;
+
+interface IChainlinkFeed {
+    function decimals() external view returns (uint8);
+    function latestRoundData() external view returns (uint80, int256, uint256, uint256, uint80);
+}
+
+interface ICurvePool {
+    function price_oracle() external view returns (uint256);
+    function last_price() external view returns (uint256);
+}
+
+contract ConvexCurvePriceFeed is IChainlinkFeed {
+    
+    IChainlinkFeed public crvToUsd = IChainlinkFeed(0xCd627aA160A6fA45Eb793D19Ef54f5062F20f33f);
+    ICurvePool public cvxCrvCrvPool = ICurvePool(0x971add32Ea87f10bD192671630be3BE8A11b8623);
+    address public gov = 0x926dF14a23BE491164dCF93f4c468A50ef659D5B;
+    uint public revertThreshold = 5000;
+
+    function decimals() external view returns (uint8){
+        return 18;
+    }
+
+    function latestRoundData() external view returns (uint80, int256, uint256, uint256, uint80){
+        (uint80 roundId,int256 crvUsdPrice,uint startedAt,uint updatedAt,uint80 answeredInRound) = crvToUsd.latestRoundData();
+        uint crvPerCvxCrv = cvxCrvCrvPool.price_oracle();
+        if(crvPerCvxCrv > 10 ** 18){
+            //1 CRV can always be traded for 1 CvxCrv, so price for CvxCrv should never be higher than the price of CRV
+            return (roundId, crvUsdPrice, startedAt, updatedAt, answeredInRound);
+        }
+        //If there's too big of a difference between the EMA price and last price
+        require(crvPerCvxCrv * revertThreshold / 10000 <= cvxCrvCrvPool.last_price(), "BELOW REVERT THRESHOLD");
```

Comment:
This is good.

It may make sense to perform this same check on upward price movements.
Even though you have a price ceiling of 1e18 already, there is theoretically still some room to manipulate price. To apply same threshold to upward price movements you could do something like this.

`require(crvPerCvxCrv * (10000 + revertThreshold) / 10000 <= cvxCrvCrvPool.last_price(), "ABOVE REVERT THRESHOLD");`

Reply:
Equal sign should be flipped right? Or it'll never be able to pass. But good call!

Reply:
Added in https://github.com/InverseFinance/FiRM/pull/35/commits/c6f365d3ad4c9bdc62908fdbd396587bbebeb314